filterpaths.py

```diff
@@ -0,0 +1,21 @@
+# -*- coding: utf-8 -*-
+
+
+def ignore_third_party_filter(report):
+
+    with open('mozilla-central/tools/rewriting/ThirdPartyPaths.txt') as f:
+        third_party_paths = [path.strip('\n') for path in f]
+
+    files = report['source_files']
+
+    filtered_report = {}
+    filtered_files = []
+    for i in files:
+        if any(i['name'].startswith(path) for path in third_party_paths):
+                continue
+        filtered_files.append(i)
+    filtered_report['source_files'] = filtered_files
```

Comment:
You could use list comprehension to do this, which is a little faster and more pythonic:
```Python
report['source_files'] = [sf for sf in report['source_files'] if any(sf['name'].startswith(path) for path in third_party_paths)
```

This way you also don't need to copy the other options, and you don't even need to return the report, since it's just modified (the caller can just do:
```Python
filterpaths.ignore_third_party_filter(report)
```